evennia/web/website/views.py

```diff
@@ -375,23 +705,22 @@ def form_valid(self, form):
         description = self.attributes.pop('desc')
         
         # Create a character
-        try:
-            character, errors = self.model.create(charname, account, description=description)
+        character, errors = self.model.create(charname, account, description=description)
```

Comment:
Reading this now I wonder if using the variable name `model` is not confusing. I know `model` is the term in Django but the `model` elsewhere in Evennia tends to refer to the actual real database model, such as `ObjectDB` or `AccountDB`. By contrast `self.model` here actually represents what we would elsewhere would call the `Typeclass` - that is, a Django proxy-model with some Evennia-specific changes. 

I'm not 100% sure about how much Django depends on `self.model` being called exactly that but probably it needs that name. I think it would help to create a get/setter property `typeclass` that uses `self.model` under the hood (for Django's benefit), but which allows you to use `self.typeclass.create` in situations like this. I just predict people will be confused by this nomenclature otherwise.

Reply:
I agree with you, but here's the other side-- Django is already a tricky and very intimidating beast for those unfamiliar with it. My goal was to avoid plastering abstractions over already complex Django concepts so people could leverage the extensive Django docs and community for their web development questions and not get frustrated trying to figure out where Django ends and Evennia begins.

We may call them Typeclasses because we can, but at the end of the day Typeclassed proxy models are still just models as far as Django and its CBVs are concerned. Most generic CBVs (and all tutorials involving them) expect a `model` field so changing this would break/require extension of a bunch of core classes and mixins and invalidate all existing Django tutorials by deviating from standard Django practice.

There are other inherent nomenclature conflicts that have no clear resolution either; Sessions as Evennia knows them have nothing to do with the Django Sessions framework, for example. I've tried to navigate this divide as delicately as I can with extensive comments ("this is a common Django thing" / "this is something special we have to do for Evennia" / "this Django thing has the same name as this Evennia thing but they are completely different") but in the end it might be safer to just accept that there's a way Evennia does things and a way Django does things and as long as we understand and document what those differences are, we don't need to overcomplicate an already-quite-complicated framework-upon-a-framework.

I'm not invested in it either way (I mean, *I* know what I'm doing), but in my opinion I don't think it does new developers any favors to force them to learn non-standard Django abstractions for no material benefit, especially when they're already complaining about its inaccessibility. But that's just my opinion, and I'm willing to admit I don't actually know what I'm doing after all :)

Reply:
I suppose the `EvenniaForm` could act as a base class for the form case of `self.model`. As for the views, there are already Evennia-specific mixins `CharacterMixin`, `AccountMixin` etc. Rather than have them all be their own bases, why not have them inherit from common base `EvenniaMixin`?

Reply:
Coincidentally, I really like your idea of likening views to Commands conceptually. I think it would help MU* users understand Django's role better if one were to present them next to each other - on one hand people entering MU commands that get handled by Commands and on the other hand people requesting web pages that get served by Django views ("Web Commands").

Reply:
> Coincidentally, I really like your idea of likening views to Commands conceptually. I think it would help MU* users understand Django's role better if one were to present them next to each other - on one hand people entering MU commands that get handled by Commands and on the other hand people requesting web pages that get served by Django views ("Web Commands").

Heh thanks. They really are truly alike-- everything you do in Command.func() is equally applicable within View.get(), hence my recent efforts to liberate logic from Commands and into methods accessible by both sides of the fence.